medical_compliance/medical_compliance/api/models.py

```diff
@@ -70,8 +70,23 @@ def __unicode__(self):
         return u'Measurement of type: %s, value: %s %s, from: %s' % (self.type, self.value, self.MEASUREMENT_SI_UNIT[self.type], pretty_date)
 
 
-# class MeasurementNotification(models.Model):
-#     withings_user_id = models.BigIntegerField(name='userid')
-#     startdate = models.BigIntegerField()
-#     enddate = models.BigIntegerField()
-#     measurement_type = models.IntegerField(name='appli')
\ No newline at end of file
+class WeightMeasurement(models.Model):
+    INPUT_SOURCES = (
+        ('withings', 'withings'),
+    )
+    MEASUREMENT_UNITS = (
+        ('kg', 'kg'),
+    )
+
+    user_id = models.BigIntegerField(name='user_id')
+    input_source = models.CharField(max_length=20, choices=INPUT_SOURCES)
+    measurement_unit = models.CharField(max_length=2, choices=MEASUREMENT_UNITS)
+    timestamp = models.BigIntegerField()
+    timezone = models.CharField(max_length=64)
+    value = models.FloatField()
+
+    @staticmethod
+    def get_previous_weight_measures(reference_id, no_weights):
+        return list(WeightMeasurement.objects.raw(
+            u'select * from api_weightmeasurement where timestamp <= (select timestamp from api_weightmeasurement where id=%s) order by timestamp desc limit %s' % (reference_id, no_weights)
```

Comment:
Hmm, this doesn't make use of Django's powerful ORM. Consider using `filter` and `order_by` described here: https://docs.djangoproject.com/en/1.10/ref/models/querysets/#order-by . For the subquery you could actually do 2 queries or use an unevaluated Django query set as described here: http://stackoverflow.com/a/20873877 .